### ISSUE TITLE
Logit buffer reuse in %copy-logits to reduce GC pressure

### ISSUE THREAD

**Issue by licjon** [licjon/cl-llama-cpp#106]
## Problem

`%copy-logits` (generation.lisp:524) allocates a fresh `(simple-array single-float n-vocab)` on every call:

```lisp
(let ((copy (make-array n-vocab :element-type 'single-float)))
  (dotimes (i n-vocab)
    (setf (aref copy i) (cffi:mem-aref ptr :float i)))
  copy)
```

For a 32k-vocab model this is ~128KB of consing per token. In a serving loop with `:logit-callback`, this fires on every generated token, creating significant GC pressure.

## Proposed change

Let `%invoke-logit-callback` accept (and `generate`/`generate-parallel` expose) an optional pre-allocated logit buffer. When supplied, `%copy-logits` writes into it instead of consing a fresh array.

The callback signature would become `(logits n-vocab)` as today — the caller just gets the same buffer refilled each time. Callers who need to retain a snapshot can copy it themselves, but the common case (inspect-and-discard) avoids all allocation.

Rough sketch:

```lisp
(defun %copy-logits (ctx-ptr n-vocab &optional buffer)
  (let ((dst (or buffer (make-array n-vocab :element-type 'single-float))))
    (declare (type (simple-array single-float (*)) dst))
    (let ((ptr (%llama:get-logits-ith ctx-ptr -1)))
      (dotimes (i n-vocab)
        (setf (aref dst i) (cffi:mem-aref ptr :float i))))
    dst))
```

Users opt in via a new `:logit-buffer` keyword on `generate`:

```lisp
(let ((buf (make-array (n-vocab model) :element-type 'single-float)))
  (generate ctx "prompt"
            :logit-buffer buf
            :logit-callback (lambda (logits n-vocab)
                              ;; logits is the same buf each time
                              (do-something logits))))
```

## Impact

Eliminates ~128KB/token of allocation in the hottest callback path. Matters most for serving loops and custom sampling that inspects logits every token.
